tests/lib/cli-engine/file-enumerator.js

```diff
@@ -565,7 +566,33 @@ describe("FileEnumerator", () => {
             });
         });
 
-        describe("if contains symbolic links", async () => {
+        const shouldSkipSymlinkTest = (() => {
+            if (process.platform !== "win32") {
+                return false;
+            }
+
+            const root = path.join(os.tmpdir(), "eslint/windows-symlink-test");
+
+            fs.mkdirSync(root, { recursive: true });
+            fs.writeFileSync(path.join(root, "file"), "");
+            try {
+                fs.symlinkSync(path.join(root, "file"), path.join(root, "symlink"), "file");
+            } catch (error) {
+                if (error.code === "EPERM" && error.syscall === "symlink") {
+                    // eslint-disable-next-line no-console -- Display help information
+                    console.warn(chalk.yellow("'if contains symbolic links' test can not run on this machine, active 'Developer Mode' may help.\nSee https://learn.microsoft.com/en-us/windows/apps/get-started/enable-your-device-for-development#activate-developer-mode"));
+                    return true;
+                }
+                throw error;
+            } finally {
+                fs.rmSync(root, { recursive: true, force: true });
+            }
+
+            return false;
+
+        })();
+
+        (shouldSkipSymlinkTest ? describe.skip : describe)("if contains symbolic links", () => {
```

Comment:
I'm not sure if it's a good idea to just skip the test, as it could silently stop working in CI. Perhaps we could rather wrap this in try-catch, log the help info (or add it to the error message), and rethrow the error:

https://github.com/eslint/eslint/blob/9c5c6ee7734c6a5918a5983d4f2bd971ca3225a8/tests/lib/cli-engine/file-enumerator.js#L582-L583

Reply:
Couldn't we just check if `process.platform === "win32"` and skip if true?

Reply:
> Perhaps we could rather wrap this in try-catch, log the help info (or add it to the error message), and rethrow the error:

I'm not sure everyone can/want make it work on Windows. How about only allow skip if `!process.env.CI`?

> Couldn't we just check if process.platform === "win32" and skip if true?

AFAIK, not every version of Windows has this problem, what if there are bugs on Windows in this test, and it seems to be working on CI, more tests are always good.